plugins/linux-spi-lpc/fu-plugin-linux-spi-lpc.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+
+#define FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR	"/sys/kernel/security/spi"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+static void
+fu_plugin_add_security_attr_bioswe (FuPlugin *plugin, FuSecurityAttrs *attrs)
+{
+	gsize bufsz = 0;
+	g_autofree gchar *buf = NULL;
+	g_autofree gchar *fn = NULL;
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	/* create attr */
+	attr = fwupd_security_attr_new ("org.kernel.BIOSWE");
+	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_CRITICAL);
+	fwupd_security_attr_set_name (attr, "SPI");
+	fu_security_attrs_append (attrs, attr);
+
+	/* load file */
+	fn = g_build_filename (FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR, "bioswe", NULL);
+	if (!g_file_get_contents (fn, &buf, &bufsz, &error_local)) {
+		g_warning ("could not open %s: %s", fn, error_local->message);
+		fwupd_security_attr_set_result (attr, "Could not open file");
```

Comment:
This errors as:

> "BIOS SPI Region: Could not open file"

So I think to make this more user friendly in the case of missing kernel, perhaps look for the directory first (IE `/sys/kernel/security/spi`) and if that fails, then indicate:

> "Kernel support for reading SPI security settings not present"

(or something similar)

Reply:
Should this fail HSI:1 or mark it as tainted?

Reply:
I think we should fail it.  This is probably the most important basic firmware test.
And I realize that means that `fwupdtool security` effectively won't work until at least kernel 5.9 at this point (or a distro backports the kernel support), but I think it's the right trade off.

Reply:
Okay, agreed. I've pushed something.

Reply:
LGTM now